R/teal_slice.R

```diff
@@ -340,6 +337,38 @@ is.teal_slices <- function(x) { # nolint
 }
 
 
+# JSON utils ------------------------------------------------------------------------------------------------------
+
+to_json <- function(x) {
+  if (!is.null(x$selected)) x$selected <- I(x$selected)
+  if (!is.null(x$choices)) x$choices <- I(x$choices)
+
+  x_json <- jsonlite::toJSON(x, pretty = TRUE, auto_unbox = TRUE, digits = 16, null = "null")
+  strsplit(x_json, split = "\n")[[1]]
+}
+
+trim_json <- function(json, nchar) {
+  json_t <- substr(json, 1, nchar)
+  substr(json_t, nchar - 2, 40) <- "..."
+  json_t
+}
```

Comment:
This function now is more universal 👍 
```suggestion
trim_character <- function(x, nchar) {
  json_t <- substr(x, 1, nchar)
  substr(x, nchar - 2, 40) <- "..."
  x
}
```